### ISSUE TITLE
kerberos constrained delegation not working

### ISSUE THREAD

**Issue by dariusan** [AgNO3/jcifs-ng#116]
There seem to be problems regarding  "constrained" ticket delegation resulting in an error: "No valid credentials provided (Mechanism level: Failed to find any Kerberos tgt)" whereas everything works fine with "unconstrained" ticket delegation. Is there a fix?

**Comment by mbechler** [comment]
I suppose you are talking about S4U2Self there, right?
I think what that would need is the ability to pass your impersonated GSSCredential down to the authentication context.

**Comment by dariusan** [comment]
Yes, the existing jcifs-ng code works already with "unconstrained delegation", since it can access ticket granting tickets (tgt). Java 8 and above support also "constrained delegation" where no "tgt" are given to the service anymore. I have an ugly but working patch of Kerb5Context where I add the GSSCredential when creating a GSSContext and jcifs-ng can access file shares successfully with "constrained delegation", i.e. impersonated users.

**Comment by arvindmaan** [comment]
It would be nice to have this enhancement through library support for kerberos S4U2Self and S4U2Proxy.

**Comment by arvindmaan** [comment]
@dariusan  i am trying to achieve S4U2Proxy, tried multiple approaches but no luck, ending up all kind of errors.
 KrbException: KDC cannot accommodate requested option (13)

 if feasible can you please share patch to fix constrained delegation issue ?

**Comment by mbechler** [comment]
Happy to incooperate the necessary changes, if someone can share the details on what information needs to be passed down and which GSSAPI calls are required.
